test/dist_autograd_test.py

```diff
@@ -689,6 +689,22 @@ def test_backward_node_failure(self):
                 # Kill all other nodes.
                 sys.exit(0)
 
+    @dist_init(clean_shutdown=False)
+    def test_context_cleanup_timeout(self):
+        global known_context_ids
+        dst_ranks = {rank for rank in range(self.world_size) if rank != self.rank}
+        with dist_autograd.context() as context_id:
+            t1 = torch.rand((3, 3), requires_grad=True)
+            t2 = torch.rand((3, 3), requires_grad=True)
+
+            time.sleep(300);
```

Comment:
Will this test sleep for 5 minutes every time it runs? If so, it seems like it would reduce the completion of our test suite by 5 mins every time, and also reduce developer velocity when running the tests locally. Would there be a better way to test this? One possibility might be to make the `kContextTimeout` in C++ configurable from python, with a good default, then set it to something small and sleep for a bit longer than that.